.github/workflows/ci.yml

```diff
@@ -0,0 +1,117 @@
+        with:
+          java-version: ${{ matrix.java }}
+
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ matrix.java }}-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-${{ matrix.java }}-
+
+      - name: Environment info
+        run: |
+          ./gradlew --version
+          echo ANDROID_SDK_ROOT: ${ANDROID_SDK_ROOT}
+          ${ANDROID_SDK_ROOT}/tools/bin/sdkmanager --version
+          echo JAVA_HOME: ${JAVA_HOME}
+          java -version
+          javac -version
+
+      - name: Build
+        run: 'source ~/.jdk7_home && ./gradlew clean build'
+
+      - name: Check
+        run: 'source ~/.jdk7_home && ./gradlew clean check'
+
+      - name: Cleanup Gradle cache
+        # Recommended by https://docs.github.com/en/actions/guides/building-and-testing-java-with-gradle
+        run: |
+          rm -f ~/.gradle/caches/modules-2/modules-2.lock
+          rm -f ~/.gradle/caches/modules-2/gc.properties
+
+  release:
+    runs-on: ubuntu-20.04
+    # It would be nice to run this as part of the build job, since it would be
+    # faster and have less duplicated Yaml, but we will add at least JDK 11 to
+    # the matrix soon, and it would not be possible to check for all matrix
+    # results before publishing if this were just another step in that job.
+    needs: build
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Java 7
+        uses: actions/setup-java@v1
+        with:
+          java-version: 7
+
+      - name: Capture JDK7_HOME
+        run: echo "export JDK7_HOME=\"$JAVA_HOME\"" > ~/.jdk7_home
+
+      - name: Set up Java 8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+
+      - name: Cache Gradle packages
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-8-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-8-
+
+      - name: Release
+        env:
+          SECRING_GPG_IV: ${{ secrets.SECRING_GPG_IV }}
+          SECRING_GPG_KEY: ${{ secrets.SECRING_GPG_KEY }}
+          GPG_KEY_ID: ${{ secrets.GPG_KEY_ID }}
+          GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
+          ENCRYPTED_GPG_KEY_LOCATION: ./.github/secring.gpg.enc
+          GPG_KEY_LOCATION: ./.github/secring.gpg
+          NEXUS_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+          NEXUS_PASSWORD: ${{ secrets.NEXUS_PASSWORD }}
+        run: |
+          openssl enc -aes-256-cbc -K "$SECRING_GPG_KEY" -iv "$SECRING_GPG_IV" -in "$ENCRYPTED_GPG_KEY_LOCATION" -out "$GPG_KEY_LOCATION" -d &&
+          source ~/.jdk7_home &&
+          ./.github/release.sh
```

Comment:
What do you think about having a specific build step in the release job, instead of executing it as a dependency of the publish task?

Reply:
The build task runs in the `build` matrix job (currently using only java 8, but the next PR will add java 11). That's a different job and build environment from the release, which is different from Travis where build and release where running in the same build environment. But in the Travis case the release happened after `clean` and `check`, so it was also relying on the task dependencies to get everything built for publishing.

Reply:
Yes, sorry, what i meant is if it would be worth having a specific step in the release to make the specific build task that is going to be released. I was thinking that maybe if you have to take a look at the output, having that explicit step could be better than looking at the output of the `release.sh` script.